### ISSUE TITLE
Formly Form is causing components to be re-rendered when fields are focused, blurred, entered, etc.

### ISSUE THREAD

**Issue by PaddingtonTheBear** [ngx-formly/ngx-formly#2227]
**Description**
Clicking in and out of a field, or inputting text, is causing re-creation of the entire component that a formly-form is in.

**Minimal Reproduction**

https://stackblitz.com/edit/ngx-formly-ui-material-sxvbuj?file=src%2Fapp%2Fapp.component.html

I created a ```rerenderTriggerTest``` component to show the issue. You'll see the ```triggerTest()``` method gets fired twice every time a field is clicked into, clicked out of, or data is input. 

You can see the spam in the console, or on the private ```rerenderCount```.

The ```triggerTest()``` method is running twice when clicking in or out, despite the triggerTest() being outside of the ```<form>```!

Appears to be occurring on all the various UI flavors (material, etc.)

**Your Environment**

- Angular version: 9.1.0
- Formly version: 5.1.15

Occurs in Angular 8 and Formly 5.1.11 from the stackblitz example.

**Additional context**
This is causing a massive issue for my use case, where I have a formly form inside of a <div *ngFor>. Every time I click into a field, it is causing the <div *ngFor> to occur again, which locks my app because it is re-rendering everything. Even with the simple case in the example, it is trigger the method call quite a bit.

**Comment by aitboudad** [comment]
The re-rendered is triggered by zone.js whenever a user interaction occurs and not related to Formly (except the provided example doesn't illustrate the real issue).

> it is causing the <div *ngFor> to occur again

use `trackBy`

**Comment by PaddingtonTheBear** [comment]
@aitboudad Thank you for the response. I guess I jumped the gun here a little bit as I am seeing strange behavior in my app and incorrectly traced it back to the formly component I introduced. I'm likely doing things incorrectly with how I'm dynamically generating the ```FormlyFieldConfig``` array, so it's causing infinite recursion on the dynamic generation whenever the zone change is called.

I had tried adding a normal input field in my app and it wasn't causing the issue in my app, so I thought it was an issue with Formly. I added a regular input to the stackblitz example and you're right, that is normal zone.js change detection that I missed.

Apologies for the mistaken bug report!
